acceptance/tests/datadog/datadog_test.go

```diff
@@ -0,0 +1,428 @@
+
+	datadogOperatorHelmValues := map[string]string{
+		"replicaCount":     "1",
+		"image.tag":        datadog.DefaultHelmChartVersion,
+		"image.repository": "gcr.io/datadoghq/operator",
+	}
+
+	releaseName := helpers.RandomName()
+	datadogOperatorRelease := datadog.OperatorReleaseName
+
+	// Install the consul cluster in the default kubernetes ctx.
+	consulCluster := consul.NewHelmCluster(t, helmValues, ctx, cfg, releaseName)
+	consulCluster.Create(t)
+
+	// Deploy Datadog Agent via Datadog Operator and apply dogstatsd overlay
+	datadogNamespace := helmValues["global.metrics.datadog.namespace"]
+	logger.Log(t, fmt.Sprintf("deploying datadog-operator via helm | namespace: %s | release-name: %s", datadogNamespace, datadogOperatorRelease))
+	datadogCluster := datadog.NewDatadogCluster(t, ctx, cfg, datadogOperatorRelease, datadogNamespace, datadogOperatorHelmValues)
+	datadogCluster.Create(t)
+
+	logger.Log(t, fmt.Sprintf("applying datadog openmetrics patch to datadog-agent | namespace: %s", datadogNamespace))
+	k8s.DeployKustomize(t, ctx.KubectlOptionsForNamespace(datadogNamespace), cfg.NoCleanupOnFailure, cfg.NoCleanup, cfg.DebugDirectory, "../fixtures/cases/datadog-openmetrics")
+	k8s.WaitForAllPodsToBeReady(t, ctx.KubernetesClient(t), datadogNamespace, "agent.datadoghq.com/component=agent")
+
+	// Retrieve datadog-agent pod name for exec
+	podList, err := ctx.KubernetesClient(t).CoreV1().Pods(datadogNamespace).List(context.Background(), metav1.ListOptions{LabelSelector: "agent.datadoghq.com/component=agent"})
+	require.NoError(t, err)
+	require.Len(t, podList.Items, 1)
+	ddAgentName := podList.Items[0].Name
+
+	// Check the dogstats-stats of the local cluster agent to see if consul metrics
+	// are being seen by the agent
+	logger.Log(t, fmt.Sprintf("retrieving datadog-agent control api auth token from pod %s", ddAgentName))
+	bearerToken, err := k8s.RunKubectlAndGetOutputE(t, ctx.KubectlOptionsForNamespace(datadogNamespace), "exec", "pod/"+ddAgentName, "-c", "agent", "--", "cat", "/etc/datadog-agent/auth_token")
+	// Retry because sometimes the merged metrics server takes a couple hundred milliseconds
+	// to start.
+	logger.Log(t, fmt.Sprintf("scraping datadog-agent /agent/status endpoint | auth-token: %s", bearerToken))
+	var metricsOutput string
+	retry.RunWith(&retry.Counter{Count: 20, Wait: 2 * time.Second}, t, func(r *retry.R) {
+		metricsOutput, err = k8s.RunKubectlAndGetOutputE(t, ctx.KubectlOptionsForNamespace(datadogNamespace), "exec", "pod/"+ddAgentName, "-c", "agent", "--", "curl", "--silent", "--insecure", "--show-error", "--header", fmt.Sprintf("authorization: Bearer %s", bearerToken), "https://localhost:5001/agent/status")
+		require.NoError(r, err)
+	})
+	var root Root
+	err = json.Unmarshal([]byte(metricsOutput), &root)
+	require.NoError(t, err)
+	for _, check := range root.RunnerStats.Checks.Openmetrics {
+		if strings.Contains(check.CheckID, "consul") {
+			require.Equal(t, ``, check.LastError)
+			break
+		}
+		continue
+	}
```

Comment:
Not strictly required, but I wonder if we can refactor most of these test cases into function that takes in dd Helm Values, a fixture name, and a validation callback. Those seem like the only differences at first glance.

Reply:
For sure! Yeah, this definitely could be refactored that way. I want to get this merged to get the naming convention issue fixed, so I left a TODO in there for now.

https://github.com/hashicorp/consul-k8s/blob/91abb5d31b645933e40bda50c5c99748be018159/acceptance/tests/datadog/datadog_test.go#L26